### PR TITLE
Adding setgroups to process.spawn

### PR DESCRIPTION
**Pull request by carlosduclos**
Currently process can set both the uid and the gid of a process. However there are situations when more than one group is needed. This change introduces the possibility of specifying secondary groups via the setgroups(2) system call.

This change will work only on POSIX systems.

### CODE REVIEW

File: test/run-tests.c [hunk 1]
```diff
@@ -177,5 +179,14 @@ static int maybe_run_test(int argc, char **argv) {
...
   }
 
+#ifndef _WIN32
+  if (strcmp(argv[1], "spawn_helper10") == 0) {
+    int fd = open("/dev/sda", O_RDONLY);
+    ASSERT(-1 != fd);
+    close(fd);
+    return 1;
+  }
+#endif  /* !_WIN32 */
```
Comment (santigimeno) [inline test/run-tests.c]: I don't think this is portable. Not all the unixes have user `disk` defined nor `/dev/sda`. Maybe a solution would be passing the additional group_id as parameter and check the child process' secondary group is the same.

File: test/test-spawn.c [hunk 2]
```diff
@@ -1704,3 +1704,93 @@ int spawn_stdin_stdout(void) {
...
+  int r;
+  struct passwd *pw;
+  struct group *gr;
+
+  /* if not root, then this will fail */
+  /* Notice that in reality if CAP_SETGID is set, then this will work */
+  uv_uid_t uid = getuid();
+  if (uid != 0) {
+    fprintf(stderr, "spawn_setuid_setgid_setgroups skipped: not root\n");
+    return 0;
```
Comment (santigimeno) [inline test/test-spawn.c]: User `RETURN_SKIP` here

File: test/test-spawn.c [hunk 3]
```diff
@@ -1704,3 +1704,93 @@ int spawn_stdin_stdout(void) {
...
```
Comment (santigimeno) [inline test/test-spawn.c]: `RETURN_SKIP`

File: test/test-spawn.c [hunk 4]
```diff
@@ -1704,3 +1704,93 @@ int spawn_stdin_stdout(void) {
...
+  MAKE_VALGRIND_HAPPY();
+  return 0;
+}
+#endif /* !_WIN32 */
+
+#ifndef _WIN32
+TEST_IMPL(spawn_setuid_setgid_setgroups_fails) {
+  int r;
+  struct passwd *pw;
+  struct group *gr;
```
Comment (santigimeno) [inline test/test-spawn.c]: Probably the 2nd test is redundant if we're checking the additional group_id in the child process.

Comment (santigimeno) [inline test/test-spawn.c]: Maybe adding a `spawn_setgroups_fails` that tries to add additional groups running as a non-root user?

**Comment by santigimeno** [comment]
Oh, I just saw this a is a duplicate of https://github.com/libuv/libuv/pull/761.Should ew close this in favor of the former? /cc @saghul

**Comment by carlosduclos** [comment]
Sure, let's close this and focus on getting #761 ready.

**Pull request closed without merging**